lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -398,6 +398,45 @@ namespace Js
         heapArgumentsType = DynamicType::New(scriptContext, TypeIds_Arguments, objectPrototype, nullptr,
             SimpleDictionaryTypeHandler::New(scriptContext, HeapArgumentsPropertyDescriptors, _countof(HeapArgumentsPropertyDescriptors), 0, 0, true, true), true, true);
 
```

Comment:
Would it work equally well to use a SimpleTypeHandler as a PathTypeHandler? (I'm not sure it makes a big difference.)

Reply:
I thought SimpleTypeHandler didn't support accessors. It looks like calling SetAccessors would turn it into a path or dictionary type handler.

---
In reply to: [183561720](https://github.com/Microsoft/ChakraCore/pull/5032#discussion_r183561720) [](ancestors = 183561720)

Reply:
You're quite right. I didn't notice the accessor. Then, yes, we need to use a PathTypeHandler for sharing.